src/cephadm/cephadm.py

```diff
@@ -2269,6 +2269,16 @@ def _default_image(ctx: CephadmContext) -> Any:
     return cast(FuncT, _default_image)
 
 
+def executes_early(func: FuncT) -> FuncT:
+    """Decorator that indicates the command function is meant to have no
+    dependencies and no environmental requirements and can therefore be
+    executed as non-root and with no logging, etc. Commands that have this
+    decorator applied must be simple and self-contained.
+    """
+    cast(Any, func)._execute_early = True
```

Comment:
Wasn't aware you could just set an attribute on a function object like that. Good one to know about.